aten/src/ATen/native/cuda/Distributions.cu

```diff
@@ -0,0 +1,49 @@
+#include "ATen/NativeFunctions.h"
+#include "ATen/Dispatch.h"
+#include "ATen/cuda/CUDAApplyUtils.cuh"
+#include <curand.h>
+#include <curand_kernel.h>
+#include <curand_philox4x32_x.h>
+#include <utility>
+
+#include <THC/THCGeneral.h>
+#include <THC/THCHalf.h>
+#include <THC/THCApply.cuh>
+#include <THC/THCNumerics.cuh>
+#include <THC/THCTensorRandom.h>
+
+THCGenerator* THCRandom_getGenerator(THCState* state);
+
+namespace at {
+namespace native {
+
+namespace dist {
+  std::pair<uint64_t, uint64_t> get_philox_seed(Generator *gen) {
+    auto gen_ = THCRandom_getGenerator(at::globalContext().thc_state);
+    return std::make_pair(gen_->initial_seed, gen_->philox_seed_offset++);
```

Comment:
My comment about thread safety still applies. I wasn't concerned about the safety of `thrust::pair` (I just this it's simpler to use `std::pair`), it's `gen_->philox_seed_offset++` that's unsafe. I'm not exactly sure how much do we care about this, since our CPU PRNGs aren't thread safe either. Any opinions @colesbury?

Reply:
Ah, I see. Yeah, I figured since there's one THCGenerator per GPU, the failure case is multiple (Python) threads launching kernels, which seemed unlikely? Totally happy to wrap it in `std::atomic` though.

Reply:
Yeah, I think using an atomic increment would be better. There are still other parts of our RNG code that are not thread-safe, but we are working towards fixing it.